azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -284,71 +293,73 @@ def __setup_azure_slice():
-
-                CGroupConfigurator._Impl.__reload_systemd_config()
+                        CGroupConfigurator._Impl._cleanup_unit_file(unit_file)
+                        return
 
         def _reset_agent_cgroup_setup(self):
             try:
                 agent_drop_in_path = systemd.get_agent_drop_in_path()
                 if os.path.exists(agent_drop_in_path) and os.path.isdir(agent_drop_in_path):
                     files_to_cleanup = []
                     agent_drop_in_file_slice = os.path.join(agent_drop_in_path, _AGENT_DROP_IN_FILE_SLICE)
+                    files_to_cleanup.append(agent_drop_in_file_slice)
                     agent_drop_in_file_cpu_accounting = os.path.join(agent_drop_in_path,
                                                                      _DROP_IN_FILE_CPU_ACCOUNTING)
+                    files_to_cleanup.append(agent_drop_in_file_cpu_accounting)
                     agent_drop_in_file_memory_accounting = os.path.join(agent_drop_in_path,
                                                                         _DROP_IN_FILE_MEMORY_ACCOUNTING)
+                    files_to_cleanup.append(agent_drop_in_file_memory_accounting)
                     agent_drop_in_file_cpu_quota = os.path.join(agent_drop_in_path, _DROP_IN_FILE_CPU_QUOTA)
-                    files_to_cleanup.extend([agent_drop_in_file_slice, agent_drop_in_file_cpu_accounting,
-                                             agent_drop_in_file_memory_accounting, agent_drop_in_file_cpu_quota])
-                    self.__cleanup_all_files(files_to_cleanup)
-                    self.__reload_systemd_config()
+                    files_to_cleanup.append(agent_drop_in_file_cpu_quota)
+
+                    if len(files_to_cleanup) > 0:
+                        log_cgroup_info("Found drop-in files; attempting agent cgroup setup cleanup", send_event=False)
+                        self._cleanup_all_files(files_to_cleanup)
+                        self._reset_cpu_quota(systemd.get_agent_unit_name())
+
             except Exception as err:
-                logger.warn("Unable to delete Agent drop-in files while resetting the quotas: {0}".format(err))
+                logger.warn("Error while resetting the quotas: {0}".format(err))
 
         @staticmethod
-        def __reload_systemd_config():
-            # reload the systemd configuration; the new slices will be used once the agent's service restarts
+        def _enable_accounting(unit_name):
+            """
+            Enable CPU and Memory accounting for the unit
+            """
             try:
-                log_cgroup_info("Executing systemctl daemon-reload...", send_event=False)
-                shellutil.run_command(["systemctl", "daemon-reload"])
+                # since we don't use daemon-reload and drop-files for accounting, so it will be enabled with systemctl set-property
+                accounting_properties = ["CPUAccounting=yes", "MemoryAccounting=yes"]
```

Comment:
What if only one controller is not mounted?  does the set_unit_properties_run_time cmd work for the other controller which is mounted?

Reply:
good point, I haven't tested it. But we have done same thing with drop-in files. Regardless of controllers mounted we simply enabled controllers for CPU and Memory. We never seen an issue. Do you know how to unmount one controller and check this?

Reply:
I've tried in the past, but haven't been successful because cgroup has to have no processes in order to unmount. I've seen telemetry from VMs which indicate some in the fleet only have one or the other mounted.

If we're unable to mount cpu or memory, then maybe we could try to add a unit property for a controller which isn't mounted by default on our test VM? And then see if the command returns any error codes. If not, I think it's safe to leave it as is in the PR.

Reply:
I unmount the memory controller and set the MemoryAccounting=yes, seems no issue 

```
root@lisa--017-e0-n0:/sys/fs/cgroup/memory# findmnt -t cgroup --noheadings
/sys/fs/cgroup/systemd          cgroup cgroup rw,nosuid,nodev,noexec,relatime,xattr,name=systemd
/sys/fs/cgroup/cpu,cpuacct      cgroup cgroup rw,nosuid,nodev,noexec,relatime,cpu,cpuacct
/sys/fs/cgroup/devices          cgroup cgroup rw,nosuid,nodev,noexec,relatime,devices
/sys/fs/cgroup/net_cls,net_prio cgroup cgroup rw,nosuid,nodev,noexec,relatime,net_cls,net_prio
/sys/fs/cgroup/hugetlb          cgroup cgroup rw,nosuid,nodev,noexec,relatime,hugetlb
/sys/fs/cgroup/misc             cgroup cgroup rw,nosuid,nodev,noexec,relatime,misc
/sys/fs/cgroup/pids             cgroup cgroup rw,nosuid,nodev,noexec,relatime,pids
/sys/fs/cgroup/blkio            cgroup cgroup rw,nosuid,nodev,noexec,relatime,blkio
/sys/fs/cgroup/cpuset           cgroup cgroup rw,nosuid,nodev,noexec,relatime,cpuset
/sys/fs/cgroup/perf_event       cgroup cgroup rw,nosuid,nodev,noexec,relatime,perf_event
/sys/fs/cgroup/rdma             cgroup cgroup rw,nosuid,nodev,noexec,relatime,rdma
/sys/fs/cgroup/freezer          cgroup cgroup rw,nosuid,nodev,noexec,relatime,freezer
```